File: internal/providers/binance/endpoints.go
package binance

import (
	"fmt"

	"crypto-trader-vue/internal/marketdata"
)

type endpoints struct {
	rest     string
	wsPublic string
	wsMarket string
}

func endpointsFor(market marketdata.Market) (endpoints, error) {
	switch market {
	case marketdata.MarketSpot:
		return endpoints{
			rest:     "https://api.binance.com/api/v3",
			wsPublic: "wss://stream.binance.com:9443/ws",
			wsMarket: "wss://stream.binance.com:9443/ws",
		}, nil
	case marketdata.MarketFutures:
		return endpoints{
			rest:     "https://fapi.binance.com/fapi/v1",
			wsPublic: "wss://fstream.binance.com/public/ws",
			wsMarket: "wss://fstream.binance.com/market/ws",
		}, nil
	default:
		return endpoints{}, fmt.Errorf("binance does not support market %q", market)
	}
}

File: frontend/electron/utility/market-data/providers/binance/endpoints.ts
import type { Market } from '../../../../../src/types/market'

export interface BinanceEndpoints {
  rest: string
  publicWebSocket: string
  marketWebSocket: string
}

const endpoints: Record<Market, BinanceEndpoints> = {
  spot: {
    rest: 'https://api.binance.com/api/v3',
    publicWebSocket: 'wss://stream.binance.com:9443/ws',
    marketWebSocket: 'wss://stream.binance.com:9443/ws',
  },
  futures: {
    rest: 'https://fapi.binance.com/fapi/v1',
    publicWebSocket: 'wss://fstream.binance.com/public/ws',
    marketWebSocket: 'wss://fstream.binance.com/market/ws',
  },
}

export function endpointsFor(market: Market): BinanceEndpoints {
  return endpoints[market]
}
